var/spack/repos/builtin/packages/mumps/package.py

```diff
@@ -181,28 +198,54 @@ def write_makefile_inc(self):
             # hack defined by _DMAIN_COMP (see examples/c_example.c)
             makefile_conf.append("CDEFS   = -DAdd_ -DMAIN_COMP")
         else:
-            if self.compiler.name != "xl" and self.compiler.name != "xl_r":
+            if not using_xl:
                 makefile_conf.append("CDEFS   = -DAdd_")
 
         if '+shared' in self.spec:
+            # All Mumps libraries will be linked with 'inject_libs'.
+            # Usually, the rpaths will be injected by the Spack compiler
+            # wrapper, however some MPI wrappers may not call the Spack
```

Comment:
did you experience that? AFAIK at least `openmpi` has environment variables set to use Spack's compiler wrappers. Must be the same for others, if not -- it's a bug

Reply:
Yes, this is the case on one of the test machines at LLNL. For now, this is a workaround - we can remove it later if there is a more general solution.

Reply:
> Yes, this is the case on one of the test machines at LLNL

could you file a bug with details?

Reply:
LLNL's compiler wrappers on Ray and the various Power9/GPU systems are currently ignoring `OMPI_CC` (because they have their own opinions about RPATHs).  I'd rather wait and see if we can get that fixed before putting code like this in lots of packages.

Reply:
I wonder how often this issue may come up in HPC systems in general. It is probably not that rare. It may be worth adding MPI wrappers for added flexibility in Spack. What would be the drawback of that?